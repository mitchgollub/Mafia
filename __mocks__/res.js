'use strict';

const res = {};

res.status = function (statusCode) {
    this.statusCode = statusCode
    return {
        json: (object) => {
            res.json = object
        }
    }
};

res.statusCode = 0;
res.json = {};

module.exports = res;