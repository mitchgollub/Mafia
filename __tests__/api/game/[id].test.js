import res from '../../../__mocks__/res';
import gameId from '../../../pages/api/game/[id]';

test('Returns Game by Id', async () => {
  const gameRepository = {
    getGame: jest.fn().mockResolvedValue({
      code: 'AAAA',
      players: { current: [], available: [] },
    }),
  };

  const req = {
    query: {
      id: 'AAAA',
    },
  };

  const response = await gameId(req, res, gameRepository);

  expect(response.statusCode).toBe(200);
  expect(response.json).toMatchSnapshot();
});

test('Returns 400 when no game is returned', async () => {
  const gameRepository = {
    getGame: jest.fn().mockResolvedValue(null),
  };

  const req = {
    query: {
      id: 'AAAA',
    },
  };

  const response = await gameId(req, res, gameRepository);

  expect(response.statusCode).toBe(400);
});

test('Returns 500 on error', async () => {
  const gameRepository = {
    getGame: () => {
      throw new Error('Failed to connect');
    },
  };

  const req = {
    query: {
      id: 'AAAA',
    },
  };

  const response = await gameId(req, res, gameRepository);

  expect(response.statusCode).toBe(500);
});
