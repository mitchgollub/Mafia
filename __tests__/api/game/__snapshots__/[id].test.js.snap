// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns Game by Id 1`] = `
GameView {
  "code": "AAAA",
  "players": Array [],
}
`;
